csrc/multidevice/pipeline.cpp

```diff
@@ -0,0 +1,307 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <ir/builder.h>
+#include <ir/cloner.h>
+#include <ir/printer.h>
+#include <ir/utils.h>
+#include <iter_visitor.h>
+#include <multidevice/pipeline.h>
+#include <multidevice/pipeline_ir.h>
+
+namespace nvfuser {
+
+int PipelineStageDescriptor::running_unique_id_ = 0;
+
+// Utility class used for Pipeline instantiation
```

Comment:
Can you please expand the comment here? It doesn't seem immediately clear what pipeline instantiation means. I'd like to have more explicit comments about what the initial state of the given `pipeline` is (looks like only `originalFusion_` and `descriptor_` are valid at that moment). And also mention what are going to be built from the initial state.